main.c

```diff
@@ -15,14 +15,17 @@ how to use the page table and disk interfaces.
 #include <string.h>
 #include <errno.h>
 
-void page_fault_handler( struct page_table *pt, int page )
-{
-	printf("page fault on page #%d\n",page);
-	exit(1);
+
+// My understanding is that, when a page fault occurs, we need to evict data from a page.
+// To do that, we must preserve memory writes by writing the frame where the fault occurs
+// to the disk. Then, the next time the frame is loaded, the data will be preserved.
```

Comment:
I think we are evicting the data from a frame. But I agree about the writing that frame to disk. I'm just not quite sure how to write to disk without making the disk global

Reply:
I was thinking we make the disk global.